packages/query/src/index.ts

```diff
@@ -1129,6 +1129,17 @@ const generateQueryHook = async (
     isQuery = operationQueryOptions.useSuspenseQuery;
   }
 
+  // For non-GET operations, only register query OR mutation hooks, not both
+  let isMutation = verb !== Verbs.GET && override.query.useMutation;
```

Comment:
The verb is checked last, after all options have been checked, so it is best to leave this step for last.

```suggestion
  let isMutation = operationQueryOptions.useMutation || override.query.useMutation;
  
  if (verb !== Verbs.GET && isQuery) {
      isMutation = false;
  }
```